library/core-utils/src/main/java/im/vector/lib/core/utils/timer/CountUpTimer.kt

```diff
@@ -28,24 +28,25 @@ import java.util.concurrent.atomic.AtomicBoolean
 import java.util.concurrent.atomic.AtomicLong
 
 @OptIn(kotlinx.coroutines.ExperimentalCoroutinesApi::class)
-class CountUpTimer(private val intervalInMs: Long = 1_000) {
+class CountUpTimer(initialTime: Long = 0L, private val intervalInMs: Long = 1_000) {
 
     private val coroutineScope = CoroutineScope(Dispatchers.Main)
-    private val elapsedTime: AtomicLong = AtomicLong()
     private val resumed: AtomicBoolean = AtomicBoolean(false)
 
+    private val clock: Clock = DefaultClock()
+    private var lastTime: AtomicLong = AtomicLong()
+    private val elapsedTime: AtomicLong = AtomicLong(initialTime)
+
     init {
         startCounter()
     }
 
     private fun startCounter() {
-        tickerFlow(coroutineScope, intervalInMs / 10)
+        tickerFlow(coroutineScope, intervalInMs)
```

Comment:
`/ 10` was used for this reason: https://github.com/vector-im/element-android/commit/c69bc126377729de1483d084a88d1aa5fbd36327

Are you sure you want to revert this?

Reply:
@bmarty The elapsed time and the current time are updated when the timer is paused & resumed (see https://github.com/vector-im/element-android/blob/941d0e5cf4c509fbc1ef6d3f29fa4242266dc74e/library/core-utils/src/main/java/im/vector/lib/core/utils/timer/CountUpTimer.kt#L58-L66), it should be more precise now and the previous trick should not be necessary anymore. Moreover, the timer does not work anymore if the interval is too low (eg. under 10 ms), the ticks are too close and it does not notify in the expected delay (it can take more than 2 seconds to notify 1 second...), so it is better to keep the real interval value and do not divide the value by ten to prevent this bad behaviour.

Reply:
Reading the code again, I think user can observe an extra delay if we keep a tick per second.
The pb is that the `tickerFlow` ticks every second, and not when the UI has to be udpated.

Reply:
As discussed together IRL, @Florian14 will make a dedicated PR to improve `CountUpTimer` implementation, and make it respect its expected contract. Ideally with some tests around it.